pkg/archive/changes_unix.go

```diff
@@ -25,3 +26,11 @@ func statDifferent(oldStat *system.StatT, newStat *system.StatT) bool {
 func (info *FileInfo) isDir() bool {
 	return info.parent == nil || info.stat.Mode()&syscall.S_IFDIR != 0
 }
+
+func getIno(fi os.FileInfo) uint64 {
+	return uint64(fi.Sys().(*syscall.Stat_t).Ino)
+}
+
+func hasHardlinks(fi os.FileInfo) bool {
+	return fi.Sys().(*syscall.Stat_t).Nlink > 1
```

Comment:
i'm staring at pondering on whether there would be issues, but i think we'll be fine.
Overlayfs, will basically always be `> 1`, so the `seen[]` will grow more on that storage backend. But i'm trying to remember also when nlink would be above 1 :-\

I think we'll be fine